Refactor: Clean up and improvements of generic tests

Tan-Thai opened the pull request:
# Summary
Refactored **all** generic test files to follow a streamlined structure

- Cleaned up majority of duplicate code
- Added comments that needs a second opinion (TODOS)
- Refactored some lines of code to be more dynamic with tests
  - Refer to `Tour` tests
-  Summarised most test-cases into a higher level syntax (Given/When/Then in a more summarised form)

## Second opinion on these things
- TODOS and potential solutions
- Tags within `keywords` file

Resolves #26 if reveiws passes.

Tan-Thai committed:
refactor: 2 tests fully refactored

Refactored tests:
- User registers a new account successfully
- User Logs in successfully

Have temporarily added "-Refactor" after each case to illustrate which i've worked on. will remove later

Tan-Thai committed:
refactor: All Generic.robot tests refactored

Refactored all tests to be on a higher level and avoid duplicate writing.
-   Added TODOS
-   Potential rework of certain tests due to unspecific focus.

Keywords + variables files have been edited BUT not cleaned up. That will be saved for when the full refactor is complete.

Tan-Thai committed:
refactor: Refactored all Generic Test Case related files

-   Added Checkout test since it was missing
-   Removed majority of duplicate code and refactored accordingly
-   Slimmed down test cases to be more readable from an outsiders perspective

berggren00 committed:
Test cases for empty card, and VIP tickets instead of regular (#28)

Tan-Thai committed:
refactor(Ticket-Types): Made choosing the ticket keyword dynamic

More or less allowing the testcase to send in a dynamic keyword as an argument. Removed `Sleeps`

berggren00 reviewed (approved):
Looks good to me 👍

berggren00 reviewed (changes requested):
Need to add more a modular solution to the date selection.

Tan-Thai committed:
Send keys variant for selecting book tour date (fix) (#33)

* feat: Added functionality for keyword pressing when selecting a calendar

- 1 test added in generic.robot starting at line 57
- The test is a duplicate of "They add a viable tour with a chosen date to the cart", with the exeption of navigating the calendar instead of typing the date into the text box

- 3 functions added in generic-keywords.robot, with a working name starting from line 132
- They are placed subsequently in order temporarily, until fully functional

- 2 functions added in variables.py, subsequently starting at line 11
- They are called in a chain to fetch the amount of days left until that day next week, taking the name of a weekday as an argument

- Currently not working, due to generic-keywords not being able to call function "get_days_left_until_day"
- Provided commentary in generic-keywords at the point the new test fails

* fix(Date Selection): Fixed date picker in dynamic calendar ui from JS (#32)

* fix(Date Selection): Fixed date picker in dynamic calendar ui from javascript

- added weekday_helper.py to resources for all things calendar calc.
    - added funcs within these that replaces old ones from variables + generic-keywords
- Commented away old code and moved to the bottom. safekeeping for the sake of history.

* style: Added clarity to keywords, removed old comments, renamed variables

---------

Co-authored-by: Daniel Kvist <daslomo@gmail.com>

berggren00 reviewed (approved):
👍
